app-android/src/androidTest/java/com/ustadmobile/port/android/view/DateRangeFragmentTest.kt

```diff
@@ -0,0 +1,156 @@
+package com.ustadmobile.port.android.view
+
+import android.content.Context
+import androidx.core.os.bundleOf
+import androidx.fragment.app.testing.FragmentScenario
+import androidx.fragment.app.testing.launchFragmentInContainer
+import androidx.navigation.fragment.findNavController
+import androidx.test.core.app.ApplicationProvider
+import com.ustadmobile.adbscreenrecorder.client.AdbScreenRecord
+import com.ustadmobile.adbscreenrecorder.client.AdbScreenRecordRule
+import com.kaspersky.kaspresso.testcases.api.testcase.TestCase
+import com.soywiz.klock.DateTime
+import com.ustadmobile.port.android.screen.DateRangeScreen
+import com.ustadmobile.test.core.impl.CrudIdlingResource
+import com.ustadmobile.test.core.impl.DataBindingIdlingResource
+import com.toughra.ustadmobile.R
+import com.ustadmobile.core.controller.DateRangePresenter
+import com.ustadmobile.core.controller.ReportEditPresenter
+import com.ustadmobile.core.generated.locale.MessageID
+import com.ustadmobile.core.networkmanager.defaultGson
+import com.ustadmobile.core.schedule.localEndOfDay
+
+import com.ustadmobile.lib.db.entities.Moment
+import com.ustadmobile.test.port.android.util.*
+import com.ustadmobile.test.rules.ScenarioIdlingResourceRule
+import com.ustadmobile.test.rules.SystemImplTestNavHostRule
+import com.ustadmobile.test.rules.UmAppDatabaseAndroidClientRule
+import com.ustadmobile.test.rules.withScenarioIdlingResourceRule
+import org.junit.Assert
+import org.junit.Rule
+import org.junit.Test
+import com.ustadmobile.core.view.UstadView
+import com.ustadmobile.lib.db.entities.DateRangeMoment
+import org.junit.Before
+
+
+@AdbScreenRecord("DateRange screen Test")
+class DateRangeFragmentTest : TestCase() {
+
+    @JvmField
+    @Rule
+    var dbRule = UmAppDatabaseAndroidClientRule()
+
+    @JvmField
+    @Rule
+    var systemImplNavRule = SystemImplTestNavHostRule()
+
+    @JvmField
+    @Rule
+    val screenRecordRule = AdbScreenRecordRule()
+
+    lateinit var context: Context
+
+    @Before
+    fun setup() {
+        context = ApplicationProvider.getApplicationContext()
+    }
+
+    @AdbScreenRecord("given Moment not present when filled then should save to database")
+    @Test
+    fun givenNoMomentPresentYet_whenFilledInAndSaveClicked_thenShouldFinishResult() {
+        val fragmentScenario = launchFragmentInContainer(themeResId = R.style.UmTheme_App) {
```

Comment:
This can go into the init block. You can use a lateinit var if you need to access it later on in the test. 

I think the advantage of using the init block is that it would log where things went wrong (e.g. if there was a crash in the init function).